src/renderer/features/operations/OperationsConfirm/AddProxy/model/confirm-model.ts

```diff
@@ -19,67 +21,118 @@ type Input = {
+const $confirmStore = restore<Input[]>(formInitiated, null).reset(formSubmitted);
 
-const $api = combine(
+const $apis = combine(
   {
     apis: networkModel.$apis,
     store: $confirmStore,
   },
   ({ apis, store }) => {
-    return store?.chain ? apis[store.chain.chainId] : undefined;
+    if (!store) return {};
+
+    return store.reduce((acc, payload) => {
+      const chainId = payload.chain.chainId;
+      const api = apis[chainId];
+
+      if (!api) return acc;
+
+      return {
+        ...acc,
+        [chainId]: api,
+      };
+    }, {} as Record<ChainId, ApiPromise>);
   },
-  { skipVoid: false },
 );
 
-const $initiatorWallet = combine(
+const $initiatorWallets = combine(
   {
     store: $confirmStore,
     wallets: walletModel.$wallets,
   },
   ({ store, wallets }) => {
-    if (!store) return null;
+    if (!store) return {};
 
-    return walletUtils.getWalletById(wallets, store.account.walletId);
+    return store.reduce<Record<number, Wallet>>((acc, storeItem, index) => {
+      const wallet = walletUtils.getWalletById(wallets, storeItem.account.walletId);
+      if (!wallet) return acc;
+
+      const id = storeItem.id ?? index;
+
+      return {
+        ...acc,
+        [id]: wallet,
+      };
+    }, {});
   },
-  { skipVoid: false },
 );
 
-const $proxiedWallet = combine(
+const $proxiedWallets = combine(
   {
     store: $confirmStore,
     wallets: walletModel.$wallets,
   },
   ({ store, wallets }) => {
-    if (!store || !store.proxiedAccount) return undefined;
+    if (!store) return {};
+
+    return store.reduce<Record<number, Wallet>>((acc, storeItem, index) => {
+      if (!storeItem.proxiedAccount) return acc;
 
-    return walletUtils.getWalletById(wallets, store.proxiedAccount.walletId);
+      const wallet = walletUtils.getWalletById(wallets, storeItem.proxiedAccount.walletId);
+      if (!wallet) return acc;
+
+      const id = storeItem.id ?? index;
+
+      return {
+        ...acc,
+        [id]: wallet,
+      };
+    }, {});
   },
-  { skipVoid: false },
 );
 
-const $signerWallet = combine(
+const $signerWallets = combine(
   {
     store: $confirmStore,
     wallets: walletModel.$wallets,
   },
   ({ store, wallets }) => {
-    if (!store) return null;
+    if (!store) return {};
+
+    return store.reduce<Record<number, Wallet>>((acc, storeItem, index) => {
+      if (!storeItem.proxiedAccount) return acc;
+
+      const wallet = walletUtils.getWalletById(wallets, storeItem.signatory?.walletId || storeItem.account.walletId);
+      if (!wallet) return acc;
+
+      const id = storeItem.id ?? index;
 
-    return walletUtils.getWalletById(wallets, store.signatory?.walletId || store.account.walletId);
+      return {
+        ...acc,
+        [id]: wallet,
+      };
+    }, {});
   },
-  { skipVoid: false },
 );
 
 export const confirmModel = {
-  $confirmStore,
-  $initiatorWallet,
-  $signerWallet,
-  $proxiedWallet,
-  $api,
+  $confirmStore: $confirmStore.map(
```

Comment:
This is barely readable, let's raise readability of this property by extracting it into combine that can occupy more space